### PR TITLE
core: propagate start limit hit from triggered unit to path unit

### PR DESCRIPTION
**Pull request by poettering**
Fixes: #16669

### CODE REVIEW

File: src/core/automount.c [hunk 1]
```diff
@@ -507,7 +507,12 @@ static void automount_trigger_notify(Unit *u, Unit *other) {
         assert(other);
 
         /* Filter out invocations with bogus state */
-        if (other->load_state != UNIT_LOADED || other->type != UNIT_MOUNT)
+        if (!IN_SET(other->load_state,
+                    UNIT_LOADED,
+                    UNIT_NOT_FOUND,
+                    UNIT_BAD_SETTING,
+                    UNIT_ERROR,
+                    UNIT_MASKED)|| other->type != UNIT_MOUNT)
```
Comment (keszybz) [inline src/core/automount.c]: Hmm, the states that are left are UNIT_STUB and UNIT_MERGED. I think it'd be cleaner at this point to invert the condition (and maybe even emit a warning if we hit that path. Wouldn't that mean some fuckup in the unit handling logic if a stub unit or a merged unit emitted a notification?).

Also, space before `||` is missing.

**Review by keszybz — CHANGES_REQUESTED** [review]
Looks like a good fix, but I think the check in the first patch looks like something that grew organically over time and should be refactored now.

**Comment by poettering** [comment]
Force pushed a new version, which upgrades these tests to assert() and turns them around. PTAL.

I think they should be asserts since it shouldn't be possible we get into the state where state changes are triggered on UNIT_STUB or UNIT_MERGED unit load states.

**Comment by keszybz** [comment]
Looks reasonable. I guess that if we ever get the logic wrong, at least it'll be easy to notice with all the asserts.

**Comment by keszybz** [comment]
bionic-i386 timed out, and semaphoreci says something about lxc. Looks unrelated.

**Pull request merged**